Resources/config/doctrine/model/Client.orm.xml

```diff
@@ -9,5 +9,6 @@
         <field name="grants" type="oauth2_grant" nullable="true" />
         <field name="scopes" type="oauth2_scope" nullable="true" />
         <field name="active" type="boolean" />
+        <field name="confidential" type="boolean" />
```

Comment:
@florianajir When Laravel Passport implemented this feature they just made the secret column nullable and then `isConfidential === null !== $this->secret`

https://github.com/laravel/passport/pull/1065

I'm wondering what are the pros and cons of both approaches?

Reply:
@X-Coder264 Good point, the only pros I see is the ability to switch from public to confidential without regenerating the secret manually but on the cons the need of having a new data attribute.
It's quite the same IMO, if we go for the nullable secret then the actual code needs more changes.

Reply:
@florianajir I don't see the point of having a non confidential client with a secret as the Oauth specification says that non confidential / public clients are not capable of securely storing a secret. So having this extra column seems redundant.